notebook/static/notebook/js/notebook.js

```diff
@@ -2221,6 +2224,36 @@ define(function (require) {
     };
 
     /**
+     * Execute a single cell
+     * 
+     * @param cell - the cell to execute
+     * 
+     * If the cell can not be executed, it is added to a list of 
+     * cells pending execution.
+     */
+    Notebook.prototype.execute = function (cell) {
+        var executed = cell.execute();
+        if (!executed) {
+            this.pending_execution.push(cell)
```

Comment:
We could make the case that since the text is not retrieved until the cell is actually executed, we should just move the cell to the end when it is executed and already is pending. Does that sound better? (i.e., you can't execute, change text, then execute again in the current model if the kernel isn't set, but you can if the kernel is set and just not connected.)